packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/component_builder.rs

```diff
@@ -48,17 +48,32 @@ impl ComponentBuilder {
                 NormalizedNode::Element(elm) => elm,
                 _ => unreachable!(),
             };
-            let extending = elm.extending.clone().map(|e| e.take_resolution());
-            if extending.is_none() {
+            if elm.extending.is_none() {
                 continue;
             }
-            let ref_resolution = extending.unwrap();
+            let ref_source = elm.extending.clone().unwrap();
             let component = &builder.components[idx];
-            let referent = &builder.components[ref_resolution.node_idx];
+            let referent = &builder.components[ref_source.idx()];
 
-            match Self::determine_extending(&ref_resolution, component, referent) {
+            match Self::determine_extending(ref_source, component, referent) {
                 Ok(extending) => {
-                    builder.components[idx].set_extending(extending);
+                    let extending_component = matches!(extending, Extending::Component(..));
+                    let component_type_changed =
+                        component.get_component_type() != referent.get_component_type();
```

Comment:
If you don't need these for other things, define them just before the if statement where they're used. This lessens the cognitive burden of the reader.

Reply:
I agree. You just need to convince the Rust borrow-checker that I need to be able to move them down.

The `set_extending` is a mutable borrow, and if I move these references definitions below that line, then immutable borrow is sustained past the mutable borrow, violating the rules! I could redefine `component` and `referent` so they don't overlap if we really want them adjacent.